pkg/crc/machine/machine.go

```diff
@@ -703,9 +703,21 @@ func configureCluster(ocConfig oc.OcConfig, sshRunner *crcssh.SSHRunner, baseDom
 	return nil
 }
 
+func getProxyConfig(baseDomainName string) (*network.ProxyConfig, error) {
+	proxy, err := network.NewProxyConfig()
+	if err != nil {
+		return nil, err
+	}
+	if proxy.IsEnabled() {
+		proxy.AddNoProxy(fmt.Sprintf(".%s", baseDomainName))
+	}
+
+	return proxy, nil
+}
+
 func configProxyForCluster(ocConfig oc.OcConfig, sshRunner *crcssh.SSHRunner, sd *systemd.InstanceSystemdCommander,
 	baseDomainName, instanceIP string) (err error) {
-	proxy, err := network.NewProxyConfig()
+	proxy, err := getProxyConfig(baseDomainName)
```

Comment:
@cfergeau I don't think this helper is required, what we are doing with this helper is just adding the `base domain` to noproxy what is wrong it to add along with instanceIP?

Reply:
On the host, we don't need `instanceIP` in `NO_PROXY` as nothing should try to access the cluster directly through its IP, but always through `api.crc.testing`. And the logic of calling `network.NewProxyConfig()` and then adding the base domain name to `NO_PROXY` will be needed twice by the end of the patch series, so I moved this to a helper.

Reply:
I think this is needed, since when using the hostname, which gets translated to the IP address, you will perform a get request. which can be forced to be resolved over the local machines proxy settings.

You are adding it anyway: https://github.com/code-ready/crc/pull/902/files#diff-06d0e75b8f65eb58c0f83515f2a7217dR771 Right?

Reply:
It's added for the initial machine creation/proxy configuration. It's not added as part of `crc oc-env`